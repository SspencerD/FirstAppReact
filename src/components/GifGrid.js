
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import carga from '../images/loading.gif';

export const GifGrid = ({ category }) => {


    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>

            { loading && <img src={carga} alt='loading'></img>}




            <div className='card-grid'>

                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}

                        />

                    ))

                }


            </div>
            <hr />
        </>
    )
}
