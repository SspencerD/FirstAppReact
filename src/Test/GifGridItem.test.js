import React from 'react';
import { shallow } from 'enzyme';
import  { GifGridItem } from '../components/GifGridItem';


describe('Prueba en GifGridItem ', () => {

    const title='un titulo';
    const url='https://localhost/pip.js';
    const wrapper = shallow( <GifGridItem
        title={title} 
        url={url}
        /> );
  


    test(' Debe mostrar el shallow en consola', () => {


       
       
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title);

        
    });

    test('Debe tener una imagen al igual de la url y alt de los props', () => {


        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url);
        expect(img.prop('alt')).toBe(title);

        
    })

    test('el Div debe tener animate y algo mas', () => {

        const div = wrapper.find('div');
        const clase = div.prop('className');


        expect(clase.includes('animate__backInDown')).toBe(true);

        // animate__backInDown
       


        
    })
    
    
    

   
    
    
    
})