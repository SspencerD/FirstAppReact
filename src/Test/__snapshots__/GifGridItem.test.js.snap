// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGridItem   Debe mostrar el shallow en consola 1`] = `
<div
  className="card animate__backInDown animate__delay-3s"
>
  <img
    alt="un titulo"
    src="https://localhost/pip.js"
  />
  <p>
    un titulo
  </p>
</div>
`;
